### PR TITLE
Preload game assets

### PR DESCRIPTION
**Pull request by MiniaczQ**
Fixes: #71

### CODE REVIEW

File: src/game/audio/soundtrack.rs [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
         commands.entity(entity).despawn();
     }
-    let path = match event {
+    let source = match event {
         Soundtrack::Disable => {
             return;
         }
-        Soundtrack::Credits => "audio/soundtracks/Monkeys Spinning Monkeys.ogg",
-        Soundtrack::Gameplay => "audio/soundtracks/Fluffing A Duck.ogg",
+        Soundtrack::Credits => soundtracks.credits.clone(),
```
Comment (alice-i-cecile) [inline src/game/audio/soundtrack.rs:19]: These can be cheaper weak clones: you're already holding onto a durable handle.

File: src/game/audio/sfx.rs [hunk 2]
```diff
@@ -28,13 +25,8 @@ pub enum Sfx {
...
-fn random_step() -> &'static str {
-    [
-        "audio/sfx/step1.ogg",
-        "audio/sfx/step2.ogg",
-        "audio/sfx/step3.ogg",
-        "audio/sfx/step4.ogg",
-    ]
-    .choose(&mut rand::thread_rng())
-    .unwrap()
+fn random_step(sfxs: &SfxAssets) -> &Handle<AudioSource> {
```
Comment (alice-i-cecile) [inline src/game/audio/sfx.rs:28]: This should be a methods on `SfxAssets` :)

File: src/game/assets.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    pub fn all_loaded(&self, assets: &Assets<Image>) -> bool {
+        assets.contains(&self.ducky)
+    }
+}
+
+#[derive(Resource, Reflect)]
+pub struct SfxAssets {
+    pub button_hover: Handle<AudioSource>,
+    pub button_press: Handle<AudioSource>,
+    pub step1: Handle<AudioSource>,
```
Comment (alice-i-cecile) [inline src/game/assets.rs:24]: These make more sense as a HashSet of handles.

Comment (janhohenheim) [reply src/game/assets.rs:24]: You mean a `HashMap`, I suppose?

File: src/game/audio/sfx.rs (hunk 2 above)
Comment (janhohenheim) [reply src/game/audio/sfx.rs:28]: It seems weird to me to couple in-game SFX logic to the SFX asset struct. IMO this would be a bad precedent.

**Review by janhohenheim — CHANGES_REQUESTED** [review]
Agreed with 2/3 suggestions by Alice, LGTM otherwise :)

File: src/screen/loading.rs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+//! A credits screen that can be accessed from the title screen.
```
Comment (janhohenheim) [inline src/screen/loading.rs:1]: I think you forgot to update this comment ;)

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: src/screen/loading.rs (hunk 4 above)
Comment (MiniaczQ) [reply src/screen/loading.rs:1]: 😭

File: src/game/assets.rs [hunk 5]
```diff
@@ -1,69 +1,127 @@
...
+    }
+}
+
+impl Deref for SfxAssets {
+    type Target = HashMap<SfxAsset, Handle<AudioSource>>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.assets
     }
 }
```
[the remark this answers is not in the document]
Comment (MiniaczQ) [reply src/game/assets.rs]: I didn't know std had `Deref` derive, I'll do that

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**